src/lib/ChatCompletionRunner.ts

```diff
@@ -0,0 +1,53 @@
+import * as Core from 'openai/core';
+import {
+  type Completions,
+  type ChatCompletionMessage,
+  type ChatCompletionMessageParam,
+  type ChatCompletionCreateParams,
+  type ChatCompletionCreateParamsNonStreaming,
+} from 'openai/resources/chat/completions';
+import { type RunnableFunctions, type BaseFunctionsArgs } from './RunnableFunction';
+import {
+  AbstractChatCompletionRunner,
+  AbstractChatCompletionRunnerEvents,
+} from './AbstractChatCompletionRunner';
+
+export interface ChatCompletionRunnerEvents extends AbstractChatCompletionRunnerEvents {
+  content: (content: string) => void;
+}
+
+export type ChatCompletionFunctionRunnerParams<FunctionsArgs extends BaseFunctionsArgs> = Omit<
+  ChatCompletionCreateParamsNonStreaming,
+  'functions'
+> & {
+  functions: RunnableFunctions<FunctionsArgs>;
+};
+
+export class ChatCompletionRunner extends AbstractChatCompletionRunner<ChatCompletionRunnerEvents> {
```

Comment:
Seems fine but just curious why we have this abstract class separation?

Reply:
There are some subtle differences between `ChatCompletionStream` which does streaming but not function running and `ChatCompletionRunner` which does function running but not streaming… but tbh I think they've shrunk over time and we could perhaps get rid of this separation. Let me give it a try.

Reply:
ah, yeah, the types are different which is intentional, best to keep separate.